v2/rust/src/child_process_supervisor.rs

```diff
@@ -0,0 +1,93 @@
+use super::termination::TerminationFlag;
+use super::timeout::Timeout;
+use anyhow::{bail, Context, Result};
+use log::{error, warn};
+use std::collections::{HashMap, HashSet};
+use std::process::{Child, Command, ExitStatus};
+use std::thread::sleep;
+use std::time::Duration;
+use sysinfo::{Pid, PidExt, Process, ProcessExt, System, SystemExt};
+
+pub struct ChildProcessSupervisor<'a> {
+    pub command: Command,
+    pub timeout: u64,
+    pub termination_flag: &'a TerminationFlag,
+}
+
+impl ChildProcessSupervisor<'_> {
```

Comment:
A timeout functionality? It feels like this should already somewhere? This would be an example:
https://docs.rs/async-std/latest/async_std/future/fn.timeout.html

Reply:
Done